src/Services/PatientService.php

```diff
@@ -27,64 +29,16 @@ class PatientService extends BaseService
      */
     private $patient_picture_fallback_id = -1;
 
-    private $validator;
+    private $patientValidator;
 
     /**
      * Default constructor.
      */
     public function __construct()
     {
         parent::__construct('patient_data');
+        $this->patientValidator = new PatientValidator();
     }
-
-    // make this a comprehensive validation
-    public function validate($patient, $context, $id = null)
-    {
-        $this->validator = new Validator();
-        if ($id) {
-            $vPid = $this->validatePid($id);
-            if ($vPid->isNotValid()) {
-                return $vPid;
-            }
-        }
-
-        $this->validator->context(
-            'insert',
-            function (Validator $context) {
-                $context->required('fname', "First Name")->lengthBetween(2, 255);
-                $context->required('lname', 'Last Name')->lengthBetween(2, 255);
-                $context->required('sex', 'Gender')->lengthBetween(4, 30);
-                $context->required('DOB', 'Date of Birth')->datetime('Y-m-d');
-            }
-        );
-
-        $this->validator->context(
-            'update',
-            function (Validator $context) {
-                $context->copyContext(
-                    'insert',
-                    function ($rules) {
-                        foreach ($rules as $key => $chain) {
-                            $chain->required(false);
-                        }
-                    }
-                );
-            }
-        );
-
-        return $this->validator->validate($patient, $context);
-    }
-
-    public function validatePid($pid)
-    {
-        $this->validator->required('pid')->callback(
-            function ($value) {
-                return $this->verifyPid($value);
-            }
-        )->numeric();
-        return $this->validator->validate(['pid' => $pid]);
-    }
-
     public function setPid($pid)
```

Comment:
I will circle back and remove the accessors for the `pid` in the next PR. We can simply support those as parameters and make things a bit more stateless.